base-images-sbom-script/app/base_images_sbom_script.py

```diff
@@ -0,0 +1,136 @@
+
+    :param base_images_digests: (List) - list of base images digests, same as BASE_IMAGE_DIGESTS tekton result
+    :param is_last_from_scratch: (Boolean) - Is the last stage/base image from scratch?
+    :return: components (List) - List of dict items in which each item contains sbom data about each base image
+    """
+
+    components = []
+
+    # property_name shows whether the image was used only in the building process
+    # or if it is the final base image. If the final base image is scratch, then
+    # this is omitted, because we aren't including scratch in the sbom.
+    for index, image in enumerate(base_images_digests):
+        property_name = "konflux:container:is_builder_image:for_stage"
+        property_value = str(index)
+        if index == len(base_images_digests) - 1 and not is_last_from_scratch:
+            property_name = "konflux:container:is_base_image"
+            property_value = "true"
+
+        parsed_image = parse_image_reference_to_parts(image)
+
+        purl = PackageURL(
+            type="oci",
+            name=parsed_image.name,
+            version=parsed_image.digest,
+            qualifiers={
+                "repository_url": parsed_image.repository_without_tag,
+                "tag": parsed_image.tag,
+            },
+        )
+
+        component = {
+            "type": "container",
+            "name": parsed_image.repository_without_tag,
+            "purl": purl.to_string(),
+            "properties": [{"name": property_name, "value": property_value}],
+        }
+        components.append(component)
+
+    return components
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Updates the sbom file with base images data based on the provided files"
+    )
+    parser.add_argument("--sbom", type=pathlib.Path, help="Path to the sbom file", required=True)
+    parser.add_argument(
+        "--base-images-from-dockerfile",
+        type=pathlib.Path,
+        help="Path to the file containing base images extracted from Dockerfile via grep, sed and awk in the buildah "
+        "task",
+        required=True,
+    )
+    parser.add_argument(
+        "--base-images-digests",
+        type=pathlib.Path,
+        help="Path to the file containing base images digests."
+        " This is taken from the BASE_IMAGES_DIGEST tekton result that was generated from"
+        "the output of 'buildah images'",
+        required=True,
+    )
+    args = parser.parse_args()
+
+    return args
+
+
+def main():
+
+    args = parse_args()
+
+    with args.base_images_from_dockerfile.open("r") as f:
+        base_images_from_dockerfile = f.read().splitlines()
+
+    with args.base_images_digests.open("r") as f:
+        base_images_digests = f.read().splitlines()
+
+    is_last_from_scratch = False
+    if base_images_from_dockerfile[-1] == "scratch":
+        is_last_from_scratch = True
+
+    with args.sbom.open("r") as f:
+        sbom = json.load(f)
```

Comment:
minor suggestion for reducing the number of lines:

```python
base_images_from_dockerfile = args.base_images_from_dockerfile.read_text().splitlines()
base_images_digests = args.base_images_digests.read_text().splitlines()
sbom = json.load(args.sbom.read_bytes())
```

Reply:
Happy to use the first two lines. But the sbom part did not work out of the box, and also we are also writing to the file later, so I kept it to avoid further changes.